### ISSUE TITLE
book and code are different

### ISSUE THREAD

**Issue by brownviper** [Book-Microservices-v2/chapter03#3]
https://github.com/Book-Microservices-v2/chapter03/blob/ecf489e28d65bd4e0d28ff9c80c1f9e09b87be3b/multiplication/src/main/java/microservices/book/multiplication/challenge/ChallengeAttempt.java#L15

chapter 3 in the book mention 
Long userId
and code here is using
User user

**Comment by mechero** [comment]
Good catch! It's an error in Listing 3.8 in the book. I checked the future references to that class and they're sync'ed with the book, so it only happens here. 

Thanks a lot for reporting it! I'll document this in the README as an errata and will mention you there.

**Comment by SIdjagtap83** [comment]
Yes!
Actually going through the book i realised the same. When writing the test case to verify ChallengeAttempts we need to create a object of ChallengeAttempt class and while declaring ChallengeAttempt the userId variable is of Long data type which should be changed to User class data type.
